### DISCUSSION TITLE
Parcel support?

### DISCUSSION THREAD

**Discussion by nikuda** [vanilla-extract-css/vanilla-extract#455] [Ideas]
Are there any instructions in getting `vanilla-extract` to work with Parcel, or does a new plugin need to be written for it?

**Comment by mattcompiles** [comment]
A new plugin needs to be written in order to support parcel. It's not something we have planned ATM though. If someone created a PR we'd be interested.

**Reply by LekoArts** [reply]
@mattcompiles @markdalgleish I'm currently dipping my toes into Parcel for work and figured that writing a transformer would be a good way to learn more about Parcel. The Transformer doc itself is quite straight-forward to understand (https://parceljs.org/plugin-system/transformer/) and I have _something_ wired up:

<img width="1512" alt="Bildschirmfoto 2021-11-30 um 21 10 29" src="https://user-images.githubusercontent.com/16143594/144120515-382368f5-a085-4a27-90d6-6efb943a60f6.png">

<img width="1422" alt="Bildschirmfoto 2021-11-30 um 21 10 47" src="https://user-images.githubusercontent.com/16143594/144120572-f6af4e9b-f5be-484f-8047-bd68e5f804b2.png">

But that's obviously still far away from something working :D Can you give me some hints/directions on how I should build this?
When looking at integrations y'all already built (e.g. https://github.com/seek-oss/vanilla-extract/blob/master/packages/esbuild-plugin/src/index.ts) I don't quite get (also on a high-level) how an integration would be built.

**Reply by LekoArts** [reply]
Hi @markdalgleish - I wrote you on Twitter DMs about this, too, can you have a look? :)

**Reply by prichodko** [reply]
@LekoArts have you managed to make this work? 😄

**Reply by LekoArts** [reply]
@prichodko No, sadly I didn't receive any answer yet that would guide me into the right direction 😬

**Reply by mattcompiles** [reply]
Hey @LekoArts,

Looking at the parcel docs it sounds like we'd want to use the [returning-multiple-assets](https://parceljs.org/plugin-system/transformer/#returning-multiple-assets) approach. The `serializeVirtualCssPath` option on `processVanillaFile` receives the source for the virtual CSS files it depends on which should then each be assigned as assets with deps on the main code file. Not sure if you're still supposed to render the CSS import statements in the code or not. If not, you could just return `''` from `serializeVirtualCssPath`.

**Reply by kasperstorgaard** [reply]
Any hidden progress here recently?

**Reply by LekoArts** [reply]
@kasperstorgaard https://github.com/vanilla-extract-css/vanilla-extract/pull/855

**Reply by kasperstorgaard** [reply]
Great, thanks!
(might take it for a spin later)

**Reply by kasperstorgaard** [reply]
Works great locally 👍
